lib/widgets/events_timeline/desktop/timeline_sidebar.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:bluecherry_client/providers/settings_provider.dart';
+import 'package:bluecherry_client/utils/extensions.dart';
+import 'package:bluecherry_client/utils/tree_view/tree_view.dart';
+import 'package:bluecherry_client/widgets/device_grid/device_grid.dart';
+import 'package:bluecherry_client/widgets/events_timeline/desktop/timeline.dart';
+import 'package:bluecherry_client/widgets/events_timeline/events_playback.dart';
+import 'package:bluecherry_client/widgets/misc.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class TimelineSidebar extends StatefulWidget {
+  const TimelineSidebar({super.key, required this.timeline});
+
+  final Timeline timeline;
+
+  @override
+  State<TimelineSidebar> createState() => _TimelineSidebarState();
+}
+
+class _TimelineSidebarState extends State<TimelineSidebar> {
+  @override
+  Widget build(BuildContext context) {
+    final loc = AppLocalizations.of(context);
+
+    return Container(
+      constraints: kSidebarConstraints,
+      height: double.infinity,
+      child: Card(
+        shape: const RoundedRectangleBorder(
+          borderRadius: BorderRadiusDirectional.vertical(
+            top: Radius.circular(12.0),
+          ),
+        ),
+        margin: const EdgeInsetsDirectional.symmetric(horizontal: 4.0),
+        child: Column(children: [
+          SubHeader(loc.servers, height: 40.0),
+          Expanded(
+            child: buildTreeView(context, setState: setState),
+          ),
+          const SubHeader('Time filter', height: 24.0),
+          ListTile(
+            title: Text(SettingsProvider.instance.dateFormat
+                .format(widget.timeline.currentDate)),
+            onTap: () async {
+              final date = await showDatePicker(
+                context: context,
+                initialDate: widget.timeline.currentDate,
+                firstDate: DateTime(DateTime.now().year),
+                lastDate: DateTime.now(),
+                initialEntryMode: DatePickerEntryMode.calendarOnly,
+                currentDate: widget.timeline.currentDate,
+              );
+              debugPrint('date: $date');
+            },
+          ),
```

Comment:
The time filter doesn't do nothing by now. The backend service doesn't provide a way to fetch events based on the date. Leave it here for future usage